src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -1981,6 +1981,41 @@ private static Expression BindParameterFromProperty(ParameterInfo parameter, Mem
         type == typeof(StringValues?) ? typeof(StringValues?) :
         null;
 
+    private static Expression CreateDefaultValueExpression(object? defaultValue, Type parameterType)
+    {
+        if (defaultValue is null)
+        {
+            return Expression.Constant(null, parameterType);
+        }
+
+        var underlyingType = Nullable.GetUnderlyingType(parameterType);
+        var isNullable = underlyingType != null;
+        var targetType = isNullable ? underlyingType! : parameterType;
+        var converted = defaultValue;
+
+        // Apply a conversion for scenarios where the default value's type
+        // doesn't match the parameter type
+        if (targetType.IsEnum && defaultValue.GetType() != targetType)
+        {
+            converted = Enum.ToObject(targetType, defaultValue);
+        }
+        else if (!targetType.IsAssignableFrom(defaultValue.GetType()))
+        {
+            try
+            {
+                converted = Convert.ChangeType(defaultValue, targetType, CultureInfo.InvariantCulture);
+            }
+            catch
+            {
+                converted = targetType.IsValueType ? Activator.CreateInstance(targetType) : null;
```

Comment:
When would this happen?

Reply:
Actually, I think we can get rid of this case. The `Convert.ChangeType` can throw if the conversion is not supported (which I think our IsAssignableFrom covers) or if the conversion overflows (which I don't think can functionally happen if the compiler catches you doing something like `short foo = 523525242`.